### DISCUSSION TITLE
Kyuubi uses Kerberos authentication to obtain the principal information as the executing user

### DISCUSSION THREAD

**Discussion by huangjjfff** [apache/kyuubi#6657] [Q&A]
The configuration file for kyuubi-defaults is as follows:

kyuubi.authentication: KERBEROS
kyuubi.kinit.principal: kyuubi/[hostname@HADOOP.COM](mailto:hostname@HADOOP.COM)
kyuubi.kinit.keytab: /etc/security/keytabs/kyuubi.service.keytab
When executed by the hdp user, an error occurs during startup.
![1725270837051](https://github.com/user-attachments/assets/69f16fe2-e68a-48d2-b12d-53103ee1bb42)

Looking into the source code, it might relate to the code running in HadoopThriftAuthBridgeServer:
private val ugi = UserGroupInformation.getCurrentUser
 used as the value for the principal.

**Comment by pan3793** [comment]
You should either set up `HADOOP_CONF_DIR` properly, or copy your Hadoop configuration files into `KYUUBI_CONF_DIR` (default is `KYUUBI_HOME/conf`)
